src/lib/containers/table/TablePagination.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react'
+import { Pagination } from '@material-ui/lab'
+import { usePaginatedQueryContext } from '../QueryContext'
+import Typography from '../../utils/typography/Typography'
+
+export const TablePagination = () => {
+  const [pageSize, setPageSize] = useState(25)
+  const [page, setPage] = useState(1)
+
+  const { executeQueryRequest, getLastQueryRequest, data, goToPage } =
+    usePaginatedQueryContext()
+  const queryCount = data?.queryCount
+
+  const handlePage = (event: React.ChangeEvent<unknown>, value: number) => {
+    setPage(value)
+    goToPage(page)
+  }
+
+  const handlePageSize = (event: React.ChangeEvent<{ value: unknown }>) => {
+    const value = event.target.value as number
+    setPageSize(value)
+  }
+
+  useEffect(() => {
+    const lastQueryRequestDeepClone = getLastQueryRequest()
+    lastQueryRequestDeepClone.query.offset = (page - 1) * pageSize
+    lastQueryRequestDeepClone.query.limit = pageSize
+    executeQueryRequest(lastQueryRequestDeepClone)
```

Comment:
This could be moved up to QueryWrapper, and then `setPageSize` could be passed in the PaginatedQueryContext